libvast/src/format/arrow.cpp

```diff
@@ -50,14 +49,16 @@ writer::~writer() {
 
 caf::error writer::write(const table_slice& slice) {
   if (out_ == nullptr)
-    return ec::filesystem_error;
+    return caf::make_error(ec::filesystem_error, "invalid arrow output stream");
   if (!layout(slice.layout()))
-    return ec::unspecified;
+    return caf::make_error(ec::logic_error, "failed to update layout");
   // Get the Record Batch and print it.
   auto batch = as_record_batch(slice);
   VAST_ASSERT(batch != nullptr);
-  if (!current_batch_writer_->WriteRecordBatch(*batch).ok())
-    return ec::filesystem_error;
+  if (auto status = current_batch_writer_->WriteRecordBatch(*batch);
+      !status.ok())
+    return caf::make_error(ec::filesystem_error, "failed to write record batch",
```

Comment:
```suggestion
    return make_error(ec::filesystem_error, "failed to write record batch",
```

Reply:
We should introduce an `arrow_error` for these.

Reply:
I'd rather make this `ec::unspecified`, as the Arrow Status contains its own category. In the case of this error, it was `Invalid`.

Reply:
That's ok as well, the inner error is what matters.